### PR TITLE
Add param_group_id support

### LINKED ISSUE #445: [RDSv3] RDS changing the param_group_id should not cause a recreation
Hi there,

I just changed the paramgroup after I created the RDS.
When doing so within the OTC-Console I can change it and need to wait for a reboot.
If I do this using terraform, it will create a new RDS instance.
Is there a way around it?

### Terraform Version
```
/usr/local/bin/terraform -v
Terraform v0.11.14
+ provider.dns v2.2.0
+ provider.local v1.4.0
+ provider.null v2.1.2
+ provider.openstack v1.23.0
+ provider.opentelekomcloud v1.14.0
+ provider.template v2.1.2
```

### Affected Resource(s)
- opentelekomcloud_rds_instance_v3
- opentelekomcloud_rds_parametergroup_v3

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### Expected Behavior
Reboot instead recreate

### Actual Behavior
forces new resource

### Steps to Reproduce
1. `terraform apply`

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request

Add possibility to set `param_group_id`
Resolve: #445 

## PR Checklist

* [x] Refers to: #445
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccRdsInstanceV3_basic
--- PASS: TestAccRdsInstanceV3_basic (1199.77s)
=== RUN   TestAccRdsInstanceV3_ip
--- PASS: TestAccRdsInstanceV3_ip (674.40s)
=== RUN   TestAccRdsInstanceV3_optionalParams
--- PASS: TestAccRdsInstanceV3_optionalParams (662.14s)
=== RUN   TestAccRdsInstanceV3_backupCheck
--- PASS: TestAccRdsInstanceV3_backupCheck (671.15s)
=== RUN   TestAccRdsInstanceV3_templateConfigCheck
=== RUN   TestAccRdsInstanceV3_templateConfigCheck
--- PASS: TestAccRdsInstanceV3_templateConfigCheck (677.34s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

**Comment by outcatcher** [comment]
recheck

File: opentelekomcloud/resource_opentelekomcloud_rds_instance_v3.go [hunk 1]
```diff
@@ -166,6 +165,12 @@ func resourceRdsInstanceV3() *schema.Resource {
 				ValidateFunc: validateECSTagValue,
 			},
 			"param_group_id": {
+				Type:       schema.TypeString,
+				Optional:   true,
+				ForceNew:   true,
+				Deprecated: "Please use `configuration_template_id` instead",
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_rds_instance_v3.go]: Please add `ConflictsWith`

File: opentelekomcloud/resource_opentelekomcloud_rds_instance_v3.go [hunk 2]
```diff
@@ -166,6 +165,12 @@ func resourceRdsInstanceV3() *schema.Resource {
...
+			},
+			"configuration_template_id": {
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_rds_instance_v3.go]: It's possible to update it using configuration template apply

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: opentelekomcloud/resource_opentelekomcloud_rds_configuration_v3.go [hunk 3]
```diff
@@ -165,7 +165,7 @@ func resourceRdsConfigurationV3Read(d *schema.ResourceData, meta interface{}) er
 		return fmt.Errorf("Error retrieving OpenTelekomCloud RDS Configuration: %s", err)
 	}
 
-	d.SetId(n.Id)
+	d.SetId(n.ID)
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_rds_configuration_v3.go:168]: ID should not be set in `read`

Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_rds_configuration_v3.go:168]: ID should not be set in `read`

File: go.sum [hunk 4]
```diff
@@ -202,6 +202,8 @@ github.com/oklog/run v1.0.0 h1:Ru7dDtJNOyC66gQ5dQmaCa0qIsAUFY3sFpK1Xk8igrw=
 github.com/oklog/run v1.0.0/go.mod h1:dlhp/R75TPv97u0XWUtDeV/lRKWPKSdTuV0TZvrmrQA=
 github.com/opentelekomcloud/gophertelekomcloud v0.2.1-0.20201215153504-49b7efc9e9b2 h1:HZB6oDXnGiKg9sEjCr71+vSO9j7EX67brLDienQwmUo=
 github.com/opentelekomcloud/gophertelekomcloud v0.2.1-0.20201215153504-49b7efc9e9b2/go.mod h1:2VmHKlpPMgw4mBsmmwaEeBo0xXqAVCjHmIOomsTD4TU=
+github.com/opentelekomcloud/gophertelekomcloud v0.2.1-0.20210115210116-fd530cf59a68 h1:wE3DaTowb7ktF0zWxfzfl0mGjFHAJ+/wRgNeqPNWlN8=
```
Comment (outcatcher) [inline go.sum:203]: Please run `go mod tidy`

File: opentelekomcloud/resource_opentelekomcloud_rds_instance_v3.go [hunk 5]
```diff
@@ -676,6 +676,22 @@ func resourceRdsInstanceV3Update(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if d.HasChange("param_group_id") {
+		_, newParamGroupRaw := d.GetChange("param_group_id")
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_rds_instance_v3.go]: You can simply use `d.Get` for getting updated value

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**